sha1dc_git.c

```diff
@@ -25,10 +25,9 @@ void git_SHA1DCFinal(unsigned char hash[20], SHA1_CTX *ctx)
 /*
  * Same as SHA1DCUpdate, but adjust types to match git's usual interface.
  */
-void git_SHA1DCUpdate(SHA1_CTX *ctx, const void *vdata, unsigned long len)
+void git_SHA1DCUpdate(SHA1_CTX *ctx, const void *vdata, size_t len)
```

Comment:
This change is good, but should come first!

> Make the git default inteface be size_t. The comment

`s/inteface/interface/`

> /* We expect an unsigned long, but sha1dc only takes an int */
> is doubly wrong. Both now use size_t.

Ah, I think you missed that the comment talks about `SHA1DCUpdate()` while your change updates `git_SHA1DCUpdate()` (note the `git_` prefix of the latter!). The comment is distinctly _not_ wrong, it only needs to be adjusted to reflect your change.

However, the commit message is awfully mum about something really crucial: why on dear almost-destroyed Earth can you update the signature of `git_SHA1DCUpdate()` without having to adjust _any_ caller? Isn't that function used in the SHA1DC-specific implementation of `struct git_hash_algo`?

Alas, it is tricky. While it _is_ true that `git_SHA1_Update1` is defined the same way as for all the other SHA-1 backends:

https://github.com/git-for-windows/git/blob/06d5a16d520ea835a54f2b74b965e204c2fe9761/hash.h#L44

However, we do not use `platform_SHA1_Update()` _directly_ in `struct git_hash_algo`. Instead, we use `git_hash_sha1_update()`, which is defined thusly:

https://github.com/git-for-windows/git/blob/06d5a16d520ea835a54f2b74b965e204c2fe9761/object-file.c#L94-L97

I.e. it contains an implicit downcast from `size_t` to `unsigned long` (that is, before your patch).

Please consider the information I collected above as splendid commit message material.

Reply:
> Please consider the information I collected above as splendid commit message material.

I amended the commit message myself.